code401challenges/src/main/java/stacksandqueues/Stack.java

```diff
@@ -0,0 +1,34 @@
+package stacksandqueues;
+
+public class Stack {
+    public Node top;
+
+    public void push(int value) {
+        Node newTopNode = new Node(value);
+
+        newTopNode.next = this.top;
+        this.top = newTopNode;
+    }
+
+    public int pop() {
+        if (this.top == null) {
+            throw new NullPointerException();
+        } else {
+            Node temp = this.top;
+            int tempValue = this.top.value;
+
+            this.top = this.top.next;
+            temp.next = null;
```

Comment:
This line is extra (I think you had this in dequeue as well). It doesn't hurt to have it though!